ultralytics_yolov5.md

```diff
@@ -0,0 +1,180 @@
+background-class: hub-background
+body-class: hub
+category: researchers
+title: YOLOv5
+summary: YOLOv5 in PyTorch > ONNX > CoreML > TFLite
+image: ultralytics_yolov5_img0.jpg
+author: Ultralytics LLC
+tags: [vision, scriptable]
+github-link: https://github.com/ultralytics/yolov5
+github-id: ultralytics/yolov5
+featured_image_1: ultralytics_yolov5_img1.jpg
+featured_image_2: ultralytics_yolov5_img2.png
+accelerator: cuda-optional
+---
+
+🚀  This guide explains how to load YOLOv5 from PyTorch Hub [https://pytorch.org/hub/](https://pytorch.org/hub/)
+
+## Before You Start
+
+Start from a working python environment with **Python>=3.8** and **PyTorch>=1.6** installed, as well as `pyyaml>=5.3` for reading YOLOv5 configuration files. To install PyTorch see [https://pytorch.org/get-started/locally/](https://pytorch.org/get-started/locally/). To install dependencies:
+```bash
+$ pip install -U opencv-python pillow pyyaml tqdm  # install dependencies
+```
+
+Note that cloning the [https://github.com/ultralytics/yolov5](https://github.com/ultralytics/yolov5) repository is **not** required.
+
+
+## Load YOLOv5 From PyTorch Hub
+
+### Simple Example
+
+This simple example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes a PIL image for inference. `'yolov5s'` is the lightest and fastest YOLOv5 model. Other available models are `'yolov5m'`, `'yolov5l'`, and `'yolov5x'`, the largest and most accurate model. For details on all models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints).
+```python
+import torch
+from PIL import Image
+
+# Model
+model = torch.hub.load('ultralytics/yolov5', 'yolov5s', pretrained=True).fuse().eval()  # yolov5s.pt
+model = model.autoshape()  # for autoshaping of PIL/cv2/np inputs and NMS
+
+# Images
+torch.hub.download_url_to_file('https://github.com/ultralytics/yolov5/releases/download/v1.0/zidane.jpg', 'zidane.jpg')
+img = Image.open('zidane.jpg')  # PIL image
+
+# Inference
+prediction = model(img)  # includes NMS
+```
+
+### Detailed Example
+
+To load YOLOv5 from PyTorch Hub for **batched inference** with PIL, OpenCV, Numpy or PyTorch inputs, with results plotted, saved and printed to screen:
+```python
+import cv2
+import numpy as np
+import torch
+from PIL import Image, ImageDraw
+
+# Model
+model = torch.hub.load('ultralytics/yolov5', 'yolov5s', pretrained=True).fuse().eval()  # yolov5s.pt
+model = model.autoshape()  # for autoshaping of PIL/cv2/np inputs and NMS
+
+# Images
+for f in ['zidane.jpg', 'bus.jpg']:  # download 2 images
+    torch.hub.download_url_to_file('https://github.com/ultralytics/yolov5/releases/download/v1.0/' + f, f)
+img1 = Image.open('zidane.jpg')  # PIL image
+img2 = cv2.imread('bus.jpg')[:, :, ::-1]  # OpenCV image (BGR to RGB)
+img3 = np.zeros((640, 1280, 3))  # numpy array
+imgs = [img1, img2, img3]  # batched list of images
+
+# Inference
+with torch.no_grad():
+    prediction = model(imgs, size=640)  # includes NMS
+
+# Plot
```

Comment:
break out the plotting into a separate code block

Reply:
Done.